mask/eolearn/mask/utils.py

```diff
@@ -10,23 +10,72 @@
 This source code is licensed under the MIT license found in the LICENSE
 file in the root directory of this source tree.
 """
+import warnings
+from enum import Enum
+from functools import partial
+from typing import Any, Callable, Optional, Tuple
 
 import cv2
 import numpy as np
-
-
-def map_over_axis(data, func, axis=0):
+from PIL import Image
+
+from eolearn.core.exceptions import EODeprecationWarning
+
+
+class ResizeMethod(Enum):
+    """Methods available for spatial resizing of data."""
+
+    NEAREST = "nearest"
+    LINEAR = "linear"
+    CUBIC = "cubic"
+
+    def get_cv2_method(self, dtype: Any) -> int:
```

Comment:
You can set `dtype: Union[np.dtype, type]` like we do [here](https://github.com/sentinel-hub/eo-learn/blob/v1.0.2/core/eolearn/core/utils/common.py#L153).

Reply:
added